### PR TITLE
SWARM-1992: Enable ejb-remote security

### PR DESCRIPTION
**Pull request by heiko-braun**
NOTE: This requires @Ladicek's work on https://github.com/wildfly-swarm/wildfly-swarm/pull/892 to be merged first

### CODE REVIEW

File: fractions/javaee/ejb-remote/src/main/java/org/wildfly/swarm/ejb/remote/runtime/RemoteServiceCustomizer.java [hunk 1]
```diff
@@ -44,6 +55,28 @@ public void customize() {
...
+                List<LoginModule> loginModules = auth.subresources().loginModules();
+                if (loginModules.stream().noneMatch(lm -> "Remoting".equals(lm.code()))) {
+                    loginModules.add(0, new LoginModule("Remoting")
+                            .code("Remoting")
+                            .flag(Flag.OPTIONAL)
+                            .moduleOption("password-stacking", "useFirstPass"));
+                }
+
+                ejbInstance.get().defaultSecurityDomain("other");
+                ejbInstance.get().defaultMissingMethodPermissionsDenyAccess(true);
```
Comment (Ladicek) [inline fractions/javaee/ejb-remote/src/main/java/org/wildfly/swarm/ejb/remote/runtime/RemoteServiceCustomizer.java]: These 2 lines aren't needed.

File: docs/howto/ejb-remote-security/index.adoc [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            bob:
+              roles:
+              - admin
+  remoting:
+    http-connectors:
+      http-remoting-connector:
+        security-realm: ApplicationRealm
+----
+
+The ejb client needs to be configured to use credentials on the invocation as well:
```
Comment (Ladicek) [inline docs/howto/ejb-remote-security/index.adoc]: "ejb client" -> "EJB client"

Should also mention this file is called `wildfly-config.xml`.

File: docs/howto/ejb-remote-security/index.adoc [hunk 3]
```diff
@@ -0,0 +1,57 @@
+
+[#ejb-remote-security]
+
+= Enabling EJB-Remote Security
+
+To enable security for remote ejb invocation you need to add the `management` fraction
```
Comment (Ladicek) [inline docs/howto/ejb-remote-security/index.adoc]: "ejb" -> "EJB"

File: docs/howto/ejb-remote-security/index.adoc [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+and re-configure the `Application Realm` to enforce authentication.
```
Comment (Ladicek) [inline docs/howto/ejb-remote-security/index.adoc]: "re-configure" -> "configure"

"`Application Realm`" -> "`ApplicationRealm`"

File: docs/howto/ejb-remote-security/index.adoc [hunk 5]
```diff
@@ -0,0 +1,57 @@
+
```
Comment (Ladicek) [inline docs/howto/ejb-remote-security/index.adoc:1]: So is this a HOWTO? I originally thought that we'd add this to the `ejb-remote` fraction docs, but I agree that a HOWTO is better!

**Review by Ladicek — DISMISSED** [review]

**Comment by kenfinnigan** [comment]
I would prefer if we don't advertise Remote EJBs in upstream, as that's "old ways" as far as we're concerned.

Is this purely needed for product? If so, could the HowTo only be added there?

**Comment by Ladicek** [comment]
I'd prefer to keep the difference between upstream and downstream minimal. Could we just add the new HOWTO to a product-specific profile?

**Comment by heiko-braun** [comment]
yes we can just have the howto there. @kenfinnigan would that work for you?

**Comment by Ladicek** [comment]
My concern with removing the HOWTO here is that whenever we fork for product again, we'll forget about this.

**Comment by kenfinnigan** [comment]
+1 @heiko-braun

**Comment by heiko-braun** [comment]
test this please

**Comment by heiko-braun** [comment]
@michalszynkiewicz the prod PR was done in parallel: https://github.com/wildfly-swarm-prod/wildfly-swarm/pull/108

**Comment by Ladicek** [comment]
I just rebased, resolved the conflict and fixed the docs issue.

**Comment by Ladicek** [comment]
Want to point out one thing: we actually don't have to mess up with profiles and Maven modules etc. We just need to conditionally include the document into the master document. I believe that should be fine with you @kenfinnigan.

**Comment by kenfinnigan** [comment]
If that's possible, sure

**Comment by Ladicek** [comment]
See the files in this PR @kenfinnigan, I think it should be obvious that the EJB remote crap won't be present in community docs :-)

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**